src/components/ChatList/CommunityChannelItem.tsx

```diff
@@ -28,9 +29,12 @@ const CommunityChannelItem = (props: TopicChannelItemProps) => {
     }
   };
 
+  const backgroundColor =
+    channelType() === 'ANON_TOPIC' ? COLORS.anonTopicChannel : COLORS.signedTopicChannel;
+
   return (
     <CustomPressable onPress={onChannelPressed}>
-      <View style={styles.channelContainer}>
+      <View style={[styles.channelContainer, {backgroundColor}]}>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for determining the background color based on `channelType` is repeated in multiple components (`CommunityChannelItem.tsx`, `GroupChatChannelItem.tsx`, etc.). This violates the DRY (Don't Repeat Yourself) principle. Consider extracting this logic into a utility function that takes `channelType` as an argument and returns the appropriate color. This will improve maintainability and reduce potential errors if changes are needed in the future.

Here's a sample implementation:

```javascript
// utils/channelColor.js
import {COLORS} from '../utils/theme';

export const getChannelBackgroundColor = (channelType) => {
  switch(channelType) {
    case 'ANON_TOPIC':
      return COLORS.anonTopicChannel;
    case 'SIGNED_TOPIC':
      return COLORS.signedTopicChannel;
    // Add other cases as needed...
    default:
      return COLORS.defaultChannel; // fallback color
  }
}
```

Then, you can use this function in your component like so:

```diff
-   const backgroundColor =
-     channelType() === 'ANON_TOPIC' ? COLORS.anonTopicChannel : COLORS.signedTopicChannel;
+   const backgroundColor = getChannelBackgroundColor(channelType());
```
Remember to import the `getChannelBackgroundColor` function at the top of your file:

```javascript
import {getChannelBackgroundColor} from '../../utils/channelColor';
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->